src/dotnet/App.Maui/Platforms/iOS/AppDelegate.cs

```diff
@@ -42,6 +52,28 @@ public override void DidEnterBackground(UIApplication application)
         base.DidEnterBackground(application);
     }
 
+    [Export ("messaging:didReceiveRegistrationToken:")]
+    public void DidReceiveRegistrationToken (Firebase.CloudMessaging.Messaging messaging, string fcmToken)
+    {
+        // Monitor token generation: To be notified whenever the token is updated.
+        var token = fcmToken;
+        Log.LogDebug("OnNewToken: '{Token}'", token);
+        var appServices = IPlatformApplication.Current?.Services;
+        if (appServices == null)
+            return;
+
+        var rpcHub = appServices.GetService<RpcHub>();
+        var commander = appServices.GetService<ICommander>();
+        var sessionResolver = appServices.GetRequiredService<TrueSessionResolver>();
+        if (rpcHub != null && commander != null)
+            _ = Task.Run(async () => {
+                await rpcHub.WhenClientPeerConnected().ConfigureAwait(false);
```

Comment:
Add exception handling for background task and log an error if it occurs. May be, make sense to use our BackgrounTask.